OpenRA.Mods.Common/Traits/Radar/AppearsOnRadar.cs

```diff
@@ -18,13 +18,22 @@
 
 namespace OpenRA.Mods.Common.Traits.Radar
 {
+	public enum AppearanceType { CenterPosition, Location, OccupiedCells, EntireFootprint }
+
 	public class AppearsOnRadarInfo : ConditionalTraitInfo
 	{
-		public readonly bool UseLocation = false;
+		[Desc("Specifies position type to use for radar footprint.")]
+		public readonly AppearanceType AppearanceType = AppearanceType.OccupiedCells;
 
 		[Desc("Player stances who can view this actor on radar.")]
 		public readonly Stance ValidStances = Stance.Ally | Stance.Neutral | Stance.Enemy;
 
+		[Desc("Specifies RGB values (in hex) that should be added or subtracted from base radar color.")]
+		public readonly Color ColorModifier = Color.Black;
+
+		[Desc("Specifies whether ColorModifier should be subtracted instead of added to base radar color.")]
+		public readonly bool SubtractColorModifier = false;
```

Comment:
What is the use case for these?

Reply:
https://github.com/OpenRA/OpenRA/pull/15192/commits/ba755d2d2f7e69a602dc11843e535b88995b6a22

In the original, building radar color was darker than vehicle/infantry radar color for better visual differentiation.

Reply:
Ok, I see.  Could we simplify this down to a single float Darken field instead, using the same calculation that we do for Ramp on PlayerColorPalette? It would also be worthwhile to cache the darkened colour on creation/owner change instead of redoing it every tick.